src/Catalog/CatalogWriterBase.cs

```diff
@@ -18,8 +18,13 @@ public abstract class CatalogWriterBase : IDisposable
     {
         protected List<CatalogItem> _batch;
         protected bool _open;
+        protected string _customDomain;
 
-        public CatalogWriterBase(IStorage storage, ICatalogGraphPersistence graphPersistence = null, CatalogContext context = null)
+        public CatalogWriterBase(
```

Comment:
I was confused at first as well, but `RegistrationMakerCatalogWriter` extends this class.